src/scene/lighting/world-clusters.js

```diff
@@ -274,7 +274,8 @@ class WorldClusters {
 
         lights.forEach((light) => {
             const runtimeLight = !!(light.mask & (MASK_AFFECT_DYNAMIC | MASK_AFFECT_LIGHTMAPPED));
-            if (light.enabled && light.type !== LIGHTTYPE_DIRECTIONAL && light.visibleThisFrame && light.intensity > 0 && runtimeLight) {
+            const nonZeroSpotCone = light.type !== LIGHTTYPE_SPOT || light._outerConeAngle > 0;
+            if (light.enabled && light.type !== LIGHTTYPE_DIRECTIONAL && light.visibleThisFrame && light.intensity > 0 && runtimeLight && nonZeroSpotCone) {
```

Comment:
```suggestion
            const zeroAngleSpotlight = light.type === LIGHTTYPE_SPOT || light._outerConeAngle === 0;
            if (light.enabled && light.type !== LIGHTTYPE_DIRECTIONAL && light.visibleThisFrame && light.intensity > 0 && runtimeLight && !zeroAngleSpotlight ) {
```

Reply:
Hope you don't mind the above suggestion - just found it easier to read. 😄

Reply:
completely agree!